src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1295,6 +1380,8 @@ private class ResolvedDependencyGraphItem
             public required LibraryRangeIndex[] Path { get; set; }
 
             public required List<HashSet<LibraryDependencyIndex>> Suppressions { get; set; }
+
+            public List<int>? PrunedDependencyIndices { get; set; }
```

Comment:
```suggestion
            public HashSet<VersionRangeIndex>? PrunedDependencies { get; set; }
```

Let's talk about how maybe using `VersionRangeIndex` will be more performant.

Reply:
I think the comment is in the wrong location, but I understand the feedback, so I'll resolve this in the next commit.